Source/SimpleActions/Private/SimpleAction.cpp

```diff
@@ -86,35 +95,39 @@ void USimpleAction::StartAction(UPARAM(DisplayName = "ActionActor") AActor* NewA
 {
     TRACE_CPUPROFILER_EVENT_SCOPE(USimpleAction::StartAction);
 
+    bool Flag_Failed = false;
     UWorld* World = GetWorld();
     if (!World)
     {
         // Only run in valid worlds.
         UE_LOG(LogSimpleAction, Warning, TEXT("StartAction: Failed. Action '%s' world not valid."), *GetName());
-		return;
+		Flag_Failed = true;
     }
 
-    if (!bAllowInEditorPreview && World->IsPreviewWorld())
+    if (!Flag_Failed && !bAllowInEditorPreview && World->IsPreviewWorld())
     {
 		// This Action isn't allowed to run in editor worlds.
 		UE_LOG(LogSimpleAction, Warning, TEXT("StartAction: Failed. Action '%s' not allowed in Editor Preview World: %s."), *GetName(), LexToString(GetWorld()->WorldType));
-		return;
+        Flag_Failed = true;
     }
-    if (!bAllowInLevelEditor && World->WorldType == EWorldType::Type::Editor)
+    if (!Flag_Failed && !bAllowInLevelEditor && World->WorldType == EWorldType::Type::Editor)
     {
         // This Action isn't allowed to run in editor worlds.
         UE_LOG(LogSimpleAction, Warning, TEXT("StartAction: Failed. Action '%s' not allowed in Editor World: %s."), *GetName(), LexToString(GetWorld()->WorldType));
-        return;
+        Flag_Failed = true;
     }
 
-    
+    if(Flag_Failed) {
+	    OnActionFailed.Broadcast(this);
+    }
 
     if (bActive)
     {
         if (bAllowDoubleStart)
         {
             // Cancel the running action before starting a new one.
-            CancelAction();
+        	CancelAction();
```

Comment:
tabbing is inconsistent.

Reply:
It's really not in my local files, no sure what's happened there.